pulsar-impl/src/main/java/com/datastax/oss/pulsar/sink/codecs/PulsarCodecProvider.java

```diff
@@ -41,6 +41,12 @@ public class PulsarCodecProvider implements ConvertingCodecProvider {
             || externalJavaType.equals(GenericType.of(AbstractStruct.class)))) {
       return Optional.of(new StructToUDTCodec(codecFactory, (UserDefinedType) cqlType));
     }
+    // TODO: Generalize this part to handle Avro as well
+    // TODO: Check the behavior of the above if condition if the PulsarStruct wraps a native object.
+    if (externalJavaType.equals(GenericType.of(PulsarStruct.class))
```

Comment:
`PulsarStruct` extends `AbstractStruct`
is it necessary to add this line or is is enough to test against `GenericType.of(AbstractStruct.class) `?

Reply:
in the constructor of StructToJsonNodeCodecAdapter we are using PulsarStruct.class, maybe it is better to refer only to PulsarStruct ?